jupyterhub/app.py

```diff
@@ -954,11 +954,18 @@ def init_users(self):
             try:
                 yield gen.maybe_future(self.authenticator.add_user(user))
             except Exception:
-                # TODO: Review approach to synchronize whitelist with db
-                # known cause of the exception is a user who has already been removed from the system
-                # but the user still exists in the hub's user db
                 self.log.exception("Error adding user %r already in db", user.name)
-        db.commit()  # can add_user touch the db?
+                if self.authenticator.delete_invalid_users:
+                    self.log.warning("Deleting invalid user %r", user.name)
+                    db.delete(user)
+                else:
+                    self.log.warning(dedent("""
+                    You can set
+                        c.Authenticator.delete_invalid_users = True
+                    to automatically delete users that have been invalidated,
```

Comment:
~~have been invalidated~~  do not pass Hub database validation.

Allows users to be deleted from the external system without notifying JupyterHub and its Hub database."

Not perfect wording; yet I think when it comes to validation and deletion it's better to err on the side of being precise/specific